docs/pages/developer-guide.md

```diff
@@ -97,39 +87,42 @@ $ cd frameworks/myframework
+
+```bash
+$ ./test.sh -i
+# ./build.sh aws
+```
+
+This will deploy your framework to DC/OS as a [package](#packaging). `build.sh` creates this package and uploads it to an AWS S3 bucket that is used to make it available to a DC/OS cluster.
 
-   ```bash
-   $ ./build.sh aws
-   ```
+**Note:** Uploading packages to the private Mesosphere AWS S3 buckets (the default) requires authenticating with [maws](https://github.com/mesosphere/maws). You'll need to export the `AWS_PROFILE` given by maws inside the container prior to running `build.sh`.
 
-   You will deploy your framework to DC/OS as a
-   [package](#packaging).  `build.sh` creates this package and uploads it to an AWS S3 bucket that is used to make it available to a DC/OS cluster.
+## Installing your package
 
-1. Install your package.
+`build.sh` prints instructions for installing the package that look something like this:
 
-   `build.sh` prints instructions for installing the package that look something like this:
+```bash
+$ dcos package repo remove myframework-aws
+$ dcos package repo add --index=0 myframework-aws https://mybucket.s3.amazonaws.com/stub-universe-myframework.zip
+$ dcos package install --yes myframework
+```
 
-   ```bash
-   $ dcos package repo remove myframework-aws
-   $ dcos package repo add --index=0 myframework-aws https://mybucket.s3.amazonaws.com/stub-universe-myframework.zip
-   $ dcos package install --yes myframework
-   ```
+Navigate to the [DC/OS Services UI](https://docs.mesosphere.com/latest/gui/#services) to view the deployment.
 
-   Navigate to the [DC/OS Services UI](https://docs.mesosphere.com/latest/gui/#services) to view the deployment.
+## Uninstalling your package.
 
-1. Uninstall your package.
+```bash
+$ dcos package uninstall myframework
+```
 
-   ```
-   $ dcos package uninstall myframework
-   $ dcos node ssh --master-proxy --leader "docker run mesosphere/janitor /janitor.py -r myframework-role -p myframework-principal -z dcos-service-myframework"
-   ```
+**Note:** If you're **Note:** using DC/OS 1.9 or older you also need to run the **janitor** script. The janitor script runs inside the DC/OS cluster, cleaning up ZooKeeper state and resource reservations made by a framework.
 
-   The second command above runs the **janitor** script.  The janitor
-   script runs inside the DC/OS cluster, cleaning up ZooKeeper state
-   and resource reservations made by a framework.  DC/OS will soon
-   support uninstall hooks so this can happen automatically, but for
-   now, you must manually run the janitor script as shown above.
+```bash
+$ dcos node ssh --master-proxy --leader "docker run mesosphere/janitor /janitor.py -r myframework-role -p myframework-principal -z dcos-service-myframework"
```

Comment:
Note that these depend on the framework *name* and not the package name. That should probably be made clear. One could also link the framework cleaner documentation here.